sdk/servicebus/azure-servicebus/azure/servicebus/_common/_configuration.py

```diff
@@ -1,50 +1,51 @@
-# --------------------------------------------------------------------------------------------
+# # --------------------------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 from typing import Optional, Dict, Any
 from urllib.parse import urlparse
 
 from azure.core.pipeline.policies import RetryMode
-from .._pyamqp.constants import TransportType
-
-
-DEFAULT_AMQPS_PORT = 1571
-DEFAULT_AMQP_WSS_PORT = 443
+from .constants import DEFAULT_AMQPS_PORT, DEFAULT_AMQP_WSS_PORT, TransportType
+from .._transport._pyamqp_transport import PyamqpTransport
 
 
 class Configuration(object):  # pylint:disable=too-many-instance-attributes
     def __init__(self, **kwargs):
-        self.user_agent = kwargs.get("user_agent")  # type: Optional[str]
-        self.retry_total = kwargs.get("retry_total", 3)  # type: int
+        self.user_agent: Optional[str] = kwargs.get("user_agent")
+        self.retry_total: int = kwargs.get("retry_total", 3)
         self.retry_mode = RetryMode(kwargs.get("retry_mode", 'exponential'))
-        self.retry_backoff_factor = kwargs.get(
+        self.retry_backoff_factor: float = kwargs.get(
             "retry_backoff_factor", 0.8
-        )  # type: float
-        self.retry_backoff_max = kwargs.get("retry_backoff_max", 120)  # type: int
-        self.logging_enable = kwargs.get("logging_enable", False)  # type: bool
-        self.http_proxy = kwargs.get("http_proxy")  # type: Optional[Dict[str, Any]]
+        )
+        self.retry_backoff_max: int = kwargs.get("retry_backoff_max", 120)
+        self.logging_enable: bool = kwargs.get("logging_enable", False)
+        self.http_proxy: Optional[Dict[str, Any]] = kwargs.get("http_proxy")
 
-        self.custom_endpoint_address = kwargs.get("custom_endpoint_address")  # type: Optional[str]
-        self.connection_verify = kwargs.get("connection_verify")  # type: Optional[str]
+        self.custom_endpoint_address: Optional[str] = kwargs.get("custom_endpoint_address")
+        self.connection_verify: Optional[str] = kwargs.get("connection_verify")
         self.connection_port = DEFAULT_AMQPS_PORT
         self.custom_endpoint_hostname = None
+        self.hostname = kwargs.pop("hostname")
+        amqp_transport = kwargs.pop("amqp_transport")
 
         self.transport_type = (
             TransportType.AmqpOverWebsocket
             if self.http_proxy
             else kwargs.get("transport_type", TransportType.Amqp)
         )
         # The following configs are not public, for internal usage only
-        self.auth_timeout = kwargs.get("auth_timeout", 60)  # type: int
+        self.auth_timeout: int = kwargs.get("auth_timeout", 60)
```

Comment:
nit : most libraries use float for timeouts, so we want to use that as well. Given the [numberic tower](https://peps.python.org/pep-0484/#the-numeric-tower) even if something is annotated as `float`, `int` is acceptable
```suggestion
        self.auth_timeout: float = kwargs.get("auth_timeout", 60)
```